### PR TITLE
Move all paltform test conditional marks to `tests_mark_conditions_platform_tests.yaml`. 

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

We have a specific yaml file `tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml` to record the conditional marks of platfrom test. In this PR, we move all the conditional marks of platform test in `tests_mark_conditions.yaml` to its specific yaml file. And if the key of conditional mark is duplicate in these two files, we combine them together. 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
We have a specific yaml file `tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml` to record the conditional marks of platfrom test. In this PR, we move all the conditional marks of platform test in `tests_mark_conditions.yaml` to its specific yaml file. And if the key of conditional mark is repetitive in these two files, we combine them together. 

#### How did you do it?
Move all paltform test conditional marks to `tests_mark_conditions_platform_tests.yaml` and combine the repetitive keys.

### CODE REVIEW

**Comment by lizhijianrd** [comment]
Is it possible to add some PR check to avoid people add platform testcases in `tests_mark_conditions.yaml` in future?

**Comment by yutongzhang-microsoft** [comment]
> Is it possible to add some PR check to avoid people add platform testcases in `tests_mark_conditions.yaml` in future?

Good suggestion! Maybe we can add some checks in Pre_test Static Analysis.

**Review by xwjiang-ms — APPROVED** [review]

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml [hunk 1]
```diff
@@ -772,6 +818,23 @@ platform_tests/test_advanced_reboot.py::test_fast_reboot_from_other_vendor:
     conditions:
       - https://github.com/sonic-net/sonic-mgmt/issues/11007
 
+#######################################
+#####   test_auto_negotiation.py  #####
+#######################################
+platform_tests/test_auto_negotiation.py:
+  skip:
+    reason: "auto negotiation test highly depends on test enviroments, file issue to track and skip for now"
+    conditions: https://github.com/sonic-net/sonic-mgmt/issues/5447
```
Comment (opcoder0) [inline tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:827]: This issue is closed. Will this still be skipped?

**Review by opcoder0 — APPROVED** [review]
LGTM. Just a minor comment on the closed issue.

File: tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml (hunk 1 above)
Comment (lizhijianrd) [reply tests/common/plugins/conditional_mark/tests_mark_conditions_platform_tests.yaml:827]: If issue is closed, then testcase will not be skipped.

**Review by lizhijianrd — APPROVED** [review]

**Review by wangxin — APPROVED** [review]

**Pull request merged**